### ISSUE TITLE
uv_thread_create not robust against bad RLIMIT_STACK values (macOS)

### ISSUE THREAD

**Issue by vtjnash** [libuv/libuv#1080]
On macOS, `uv_thread_create` expects `RLIMIT_STACK` to be a valid value for `pthread_attr_setstacksize`, but that isn't necessarily true (pthread requires that the stack be specified in multiples of the page size, but rlimit does not).

Simple repro:
```
ulimit -S -s 7001
./julia -e 'getaddrinfo("localhost")'

signal (4): Illegal instruction: 4
abort at /Volumes/Lion/Users/jameson/Documents/julia/usr/lib/libjulia.0.6.0.dylib (unknown line)
uv_thread_create at /Volumes/Lion/Users/jameson/Documents/julia/deps/srccache/libuv-8d5131b6c1595920dd30644cd1435b4f344b46c8/src/unix/thread.c:88
init_once at /Volumes/Lion/Users/jameson/Documents/julia/deps/srccache/libuv-8d5131b6c1595920dd30644cd1435b4f344b46c8/src/threadpool.c:173
__pthread_once_handler at /usr/lib/system/libsystem_pthread.dylib (unknown line)
_os_once at /usr/lib/system/libsystem_platform.dylib (unknown line)
pthread_once at /usr/lib/system/libsystem_pthread.dylib (unknown line)
uv_once at /Volumes/Lion/Users/jameson/Documents/julia/deps/srccache/libuv-8d5131b6c1595920dd30644cd1435b4f344b46c8/src/unix/thread.c:239
uv__work_submit at /Volumes/Lion/Users/jameson/Documents/julia/deps/srccache/libuv-8d5131b6c1595920dd30644cd1435b4f344b46c8/src/threadpool.c:184
uv_getaddrinfo at /Volumes/Lion/Users/jameson/Documents/julia/deps/srccache/libuv-8d5131b6c1595920dd30644cd1435b4f344b46c8/src/unix/getaddrinfo.c:186
```

from https://github.com/JuliaLang/julia/issues/18818

**Comment by bnoordhuis** [comment]
What version are you using?  What you describe was fixed in #864, which was released in v1.9.1.  You can cherry-pick commit 28d160f if you need.

**Comment by vtjnash** [comment]
An older version :). That looks like a good fix for us, thanks.
